packages/@react-spectrum/searchwithin/test/SearchWithin.test.js

```diff
@@ -0,0 +1,163 @@
+  );
+}
+
+describe('SearchWithin', function () {
+  beforeAll(function () {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    act(() => jest.runAllTimers());
+  });
+
+  afterAll(function () {
+    jest.restoreAllMocks();
+  });
+
+  it('renders correctly', function () {
+    let {getAllByText, getByRole, getByLabelText} = renderSearchWithin();
+
+    let searchfield = getByLabelText('Test', {selector: 'input'});
+    expect(searchfield).toBeVisible();
+    expect(searchfield).toHaveAttribute('type', 'search');
+
+    let button = getByRole('button');
+    expect(button).toBeVisible();
+    expect(button).toHaveAttribute('aria-haspopup', 'listbox');
+
+    let label = getAllByText('Test')[0];
+    expect(label).toBeVisible();
+  });
+
+  it('can type in search and get onChange', function () {
+    let onChange = jest.fn();
+    let {getByLabelText} = renderSearchWithin({}, {onChange});
+    let searchfield = getByLabelText('Test', {selector: 'input'});
+    expect(searchfield).toHaveAttribute('value', '');
+
+    act(() => {searchfield.focus();});
+    typeText(searchfield, 'test search');
+    act(() => {searchfield.blur();});
+    expect(searchfield).toHaveAttribute('value', 'test search');
+    expect(onChange).toBeCalledTimes(11);
+  });
+
+  it('can open menu and get onChange', function () {
+    let onOpenChange = jest.fn();
+    let {getByRole} = renderSearchWithin({}, {}, {onOpenChange});
+
+    let picker = getByRole('button');
+    triggerPress(picker);
+
+    let listbox = getByRole('listbox');
+    expect(listbox).toBeVisible();
+    expect(onOpenChange).toBeCalledTimes(1);
+    expect(onOpenChange).toHaveBeenCalledWith(true);
+  });
+
+  it('searchfield and picker are labelled correctly', function () {
+    let {getByRole, getAllByText, getByLabelText} = renderSearchWithin();
+
+    let searchfield = getByLabelText('Test', {selector: 'input'});
+    let picker = getByRole('button');
+    let group = getByRole('group');
+    triggerPress(picker);
+
+    let listbox = getByRole('listbox');
+    let label = getAllByText('Test')[0];
+    expect(listbox).toHaveAttribute('aria-labelledby', label.id);
+    expect(searchfield).toHaveAttribute('aria-labelledby', label.id);
+    expect(group).toHaveAttribute('aria-labelledby', label.id);
+  });
+
+  it('isDisabled=true disables both the searchfield and picker', function () {
+    let {getByRole, getByLabelText} = renderSearchWithin({isDisabled: true});
+
+    let searchfield = getByLabelText('Test', {selector: 'input'});
+    let picker = getByRole('button');
+
+    expect(searchfield).toHaveAttribute('disabled');
+    expect(picker).toHaveAttribute('disabled');
+  });
+
+  it('autoFocus=true on searchfield will automatically focus the input', function () {
+    let {getByLabelText} = renderSearchWithin({}, {autoFocus: true});
+
+    let searchfield = getByLabelText('Test', {selector: 'input'});
+
+    expect(searchfield).toHaveFocus();
+  });
```

Comment:
Add a story to make sure both the searchField and picker are focusable?

Reply:
Added, thanks.